src/modules/_[spaceId]/HomePage/HomePage.tsx

```diff
@@ -88,50 +126,95 @@ export default function HomePage({
     : [launchCommunityButton, integrateChatButton]
 
   return (
-    <DefaultLayout>
+    <DefaultLayout
+      navbarProps={{
+        customContent: (logo, auth, colorModeToggler) => {
+          return (
+            <HomePageNavbar
+              auth={auth}
+              colorModeToggler={colorModeToggler}
+              logo={logo}
+              searchProps={{
+                search,
+                setSearch,
+                removeFocusedElement,
+                onUpClick,
+                onDownClick,
+              }}
+            />
+          )
+        },
+      }}
+    >
       {!isInIframe && <WelcomeModal />}
       <div className='flex flex-col overflow-auto'>
-        {!isInIframe && specialButtons}
-        {order.map((postId) => (
-          <ChatPreviewContainer postId={postId} key={postId} />
-        ))}
+        {!isInIframe && !search && specialButtons}
+        {searchResults.map((post, idx) => {
+          if (!post) return null
+          return (
+            <ChatPreviewContainer
+              isFocused={idx === focusedElementIndex}
+              post={post}
+              key={post.id}
+            />
+          )
+        })}
       </div>
     </DefaultLayout>
   )
 }
 
-function ChatPreviewContainer({ postId }: { postId: string }) {
-  const { data } = getPostQuery.useQuery(postId)
+function ChatPreviewContainer({
+  post,
+  isFocused,
+}: {
+  post: PostData
+  isFocused?: boolean
+}) {
   const sendEvent = useSendEvent()
   const isInIframe = useIsInIframe()
   const router = useRouter()
 
-  const content = data?.content
+  const content = post?.content
+
+  const linkTo = getChatPageLink(
+    router,
+    createSlug(post.id, { title: content?.title })
+  )
+
+  useEffect(() => {
+    if (!isFocused) return
+    const listener = (e: KeyboardEvent) => {
+      const method = isInIframe ? 'replace' : 'push'
+      if (e.key === 'Enter') {
```

Comment:
Why did you not use a hotkey hook for it?

Reply:
refactored, I forgot before